### ISSUE TITLE
Invalid Access Token

### ISSUE THREAD

**Issue by PEScorp** [globelabs/api#19]
Hi Everyone,

I have issue regarding in Access token , I already use my access token but after the next day ,It response Invalid Access Token,

Can anyone help me regarding this matter,

Thanks

**Comment by patrickjeried** [comment]
I'm having the same issue. 

<img width="461" alt="Screen Shot 2019-09-18 at 3 00 43 PM" src="https://user-images.githubusercontent.com/35409266/65123085-1c014080-da25-11e9-8c5f-3d9a080285a0.png">

**Comment by wmjuat** [comment]
What access token are you using?

**Comment by kawoki** [comment]
Just found out why it returns `Invalid access token`. The problem was, you **MUST** use the recipient's *access token* for the **POST** request. For example:

```javascript

# Request URL
https://devapi.globelabs.com.ph/smsmessaging/v1/outbound/<short code>/requests?access_token=<recipient's access token>

# Request Body
{
    "outboundSMSMessageRequest": {
        "senderAddress": "1234",  // short code suffix
        "outboundSMSTextMessage": {
            "message": "Your message goes here"
        },
        "address": "9071234567" // the recipient
    }
}
```

At the end of the day, the recipient **NEEDS** to register to your app. 

There's an instruction below your app in Globe Labs
- Go to *My Apps*
- select your app
- scroll down to see the instructions

Hope this helps.

**Comment by greenlover1991** [comment]
> Just found out why it returns `Invalid access token`. The problem was, you **MUST** use the recipient's _access token_ for the **POST** request. For example:
> 
> ```js
> # Request URL
> https://devapi.globelabs.com.ph/smsmessaging/v1/outbound/<short code>/requests?access_token=<recipient's access token>
> 
> # Request Body
> {
>     "outboundSMSMessageRequest": {
>         "senderAddress": "1234",  // short code suffix
>         "outboundSMSTextMessage": {
>             "message": "Your message goes here"
>         },
>         "address": "9071234567" // the recipient
>     }
> }
> ```
> 
> At the end of the day, the recipient **NEEDS** to register to your app.
> 
> There's an instruction below your app in Globe Labs
> 
> * Go to _My Apps_
> * select your app
> * scroll down to see the instructions
> 
> Hope this helps.

Hi, @kawoki  I'm still getting `Invalid access token` even with access_token for that subscriber. How did you fix yours?

**Steps I did:**
1. Let user register via the dialog https://developer.globelabs.com.ph/dialog/oauth/<app_id>
2. After user received confirmation code and inputted code, your callback will receive the `code` as a query parameter via GET
3. Retrieve access_token for that subscriber: `curl -X POST -d 'app_id=<app_id>' -d 'app_secret=<app_secret>' -d 'code=<code in step 2>' https://developer.globelabs.com.ph/oauth/access_token`. This returned: `{"access_token":"<subscriber's access_token>","subscriber_number":"9123456789"}`
4. Send SMS to that subscriber: `curl -H "Content-Type: application/json" -d '{"outboundSMSMessageRequest":{"outboundSMSTextMessage":{"message":"hello world"},"address":"9123456789", "senderAddress":"1234"}}' https://devapi.globelabs.com.ph/smsmessaging/v1/outbound/<short code>/requests?access_token=<subscriber's access_token>`
5. [BUG] got error `{"error":"Invalid access token"}`

**Comment by kawoki** [comment]
@greenlover1991, Try using your `short code` in `senderAddress` instead of 1234

**Comment by greenlover1991** [comment]
> greenlover1991, Try using your `short code` in `senderAddress` instead of 1234

@kawoki yup, used my `short code` instead of 1234, still got the same error

**Comment by kawoki** [comment]
@greenlover1991, not quite sure what causes your error. Try `Postman` for testing api request instead of curl. It is easier than doing everything in the terminal.

**Comment by markandrewkato** [comment]
@greenlover1991 did you manage to fix your issue? I'm also getting the same issue but I entered the correct access token and sender address data which is the last 4 digit of shortcode

**Comment by greenlover1991** [comment]
@markandrewkato no. I gave up. Better to use a different SMS API gateway.

**Comment by mcandidier** [comment]
having the same problem, @markandrewkato any luck with this issue?

**Comment by pavithramm** [comment]
Having the same problem . any updates on the fix ?
